chainermn/communicators/pure_nccl_communicator.py

```diff
@@ -106,58 +113,47 @@ def _allreduce_grad_async(self, model, stream):
             self.gpu_buffer_a.array(n_elems,
                                     dtype=allreduce_grad_dtype),
             stream=stream)
-        self._unpack_params_from_buffer(params, grad_dtype,
+
+        # unpack params from buffer A -> params
+        self._unpack_params_from_buffer(params,
                                         allreduce_grad_dtype, n_elems, stream)
 
-    def _assign_for_allreduce_grad(self, grad_dtype, allreduce_grad_dtype,
-                                   n_elems):
+    def _prepare_allreduce_pack_buffer(self, grad_dtype, allreduce_grad_dtype,
+                                       n_elems):
         allreduce_grad_n_bytes = allreduce_grad_dtype.itemsize * n_elems
         needs_sync = False
+
         if self.gpu_buffer_a.size != allreduce_grad_n_bytes:
             self.gpu_buffer_a.assign(allreduce_grad_n_bytes)
             needs_sync = True
         if self.gpu_buffer_b.size != allreduce_grad_n_bytes:
             self.gpu_buffer_b.assign(allreduce_grad_n_bytes)
             needs_sync = True
 
+        # TODO(kfukuda): What type should gpu_tmp_buffer be?
```

Comment:
I think `gpu_tmp_buffer ` is not needed.

Reply:
You are right. I will remove the code region. thanks!

Reply:
NOTE: gpu_tmp_buffer is still used by bcase, so I will remove only the code in `_prepare_allreduce_pack_buffer`.